libnavui-shield/src/main/java/com/mapbox/navigation/ui/shield/ShieldsCache.kt

```diff
@@ -109,7 +110,7 @@ internal abstract class ResourceCache<Argument, Value> {
 internal class ShieldResultCache(
     private val shieldSpritesCache: ShieldSpritesCache = ShieldSpritesCache(),
     private val shieldByteArrayCache: ShieldByteArrayCache = ShieldByteArrayCache(),
-) : ResourceCache<RouteShieldToDownload, RouteShield>() {
+) : ResourceCache<RouteShieldToDownload, RouteShield>(3) {
```

Comment:
We might want to make it much, much greater. It would be best to make a data-driven decision here - run a couple of cross-country routes (like US, Japan, Germany, and cross-Europe) and see how many shields we'd encounter during a route like this, and use that max value as a cache size with some additional buffer. We don't want to be re-downloading shields on a single route or in a single area. The `LRU` component is here to prevent over-accumulation over multiple navigation session.

I'd imagine it's going to be somewhere around tens of shields that we'd want to cache.

Reply:
Based on my testing while implementing this feature, 50 should be a good number here for cross country routes.

Reply:
I tested with routes across the U.S., Japan and Europe.  For each cache type I logged the cache misses and increased the cache size until I no longer saw the cache misses increase during navigation. I then added a little more to each value.